interfaces/builtin/desktop.go

```diff
@@ -0,0 +1,132 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopSummary = `allows running in modern Desktop Environments`
```

Comment:
Well, the idea is that we are changing how we are doing this. Please see the forum for discussion.

The original plan was to have an interface per DE (unity7, unity8, gnome-shell, plasma), per service (password-manager-service, etc) and per display server (mir, wayland, x11). Per service continues to make sense (you specify the services you want) and per display server makes sense (you don't necessarily just get mir and wayland for free-- you need to code your app to use them).

Per DE I think is the wrong approach and we really want to have only two `desktop` and `desktop-legacy`. `desktop` is anything modern with safe APIs for application isolation, and `desktop-legacy` is for the old-world "everything in the user's desktop session is trusted". By adding `desktop` we prevent people from adding new DEs as they are supported in snappy, so it makes things maintainable and future-proof going forward. Some new DE ShinyWayland comes up, we just add the accesses to `desktop` and everyone wins. (Similar to how we add kde and cinnamon DBus APIs to unity7 today).

`unity7` *is* `desktop-legacy` today, and perhaps we want to provide an alias for it, but I didn't want to conflate a name transition with this PR.